smartsim/_core/mli/infrastructure/storage/featurestore.py

```diff
@@ -24,15 +24,39 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import enum
 import typing as t
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
 
+from smartsim.error.errors import SmartSimError
 from smartsim.log import get_logger
 
 logger = get_logger(__name__)
 
 
+class ReservedKeys(str, enum.Enum):
```

Comment:
As you noted, if we put something in a backbone feature store, then instantiate a `DragonFeatureStore` with the same descriptor it will incorrectly allow a write to the reserved location. This is the entire reason for the existence of the `ReservedKeys`. 

- I believe I misused ReservedKeys in `BackboneFeatureStore` when getting property values. I now avoid using the reserved keys for anything but reserved key write checks
-  I did a quick refactor to avoid needing to use it in child classes at all